include/zephyr/mgmt/mcumgr/img_mgmt_callbacks.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Laird Connectivity
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_MCUMGR_IMG_MGMT_CALLBACKS_
+#define H_MCUMGR_IMG_MGMT_CALLBACKS_
+
+#include <img_mgmt/img_mgmt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief MCUmgr img_mgmt callback API
+ * @defgroup mcumgr_callback_api_img_mgmt MCUmgr img_mgmt callback API
+ * @ingroup mcumgr_callback_api
+ * @{
+ */
+
+/**
+ * Structure provided in the MGMT_EVT_OP_IMG_MGMT_DFU_CHUNK notification callback: This callback
+ * function is used to notify the application about a pending firmware upload packet from a client
+ * and authorise or deny it. Upload will be allowed so long as all notification handlers return
+ * MGMT_ERR_EOK, if one returns an error then the upload will be denied.
+ */
+struct img_mgmt_upload_check {
+	/** Action to take */
+	struct img_mgmt_upload_action *action;
+
+	/** Upload request information */
+	struct img_mgmt_upload_req *req;
+} __packed;
```

Comment:
Why does this need to be packed?

Reply:
On Cortex M0 which does not support unaligned memory access, if this is not packed then it faults due to it accessing non-word boundary elements

Reply:
Hmm. And why the unaligned memory access happens? `img_mgmt_upload_check` seems to be only instantiated on stack and then passed to the handlers. I'm not sure why it causes a problem.

Reply:
The original issue is here: https://github.com/zephyrproject-rtos/zephyr/issues/49066

Essentially the code accesses created objects which references objects inside of the req object. I don't have an M0 to test on if this is needed for this specific struct or not

Reply:
The original issue seemed to be casting unaligned memory to the non-packed type (so the code accessing that type might incorrectly assume that the data members are aligned). Here I can't see such a cast, but I don't have time to verify it so nevermind :)